api/src/main/java/org/openmrs/module/commonreports/reports/VaccinationReportManager.java

```diff
@@ -0,0 +1,297 @@
+				        + ") OR ( obs_group_id IN ( SELECT obs_group_id FROM obs where obs_group_id IN (  SELECT obs_group_id FROM obs where concept_id ="
+				        + inizService.getConceptFromKey("report.vaccination.vaccinationDate").getConceptId()
+				        + " and value_datetime BETWEEN :onOrAfter AND :onOrBefore ) AND concept_id ="
+				        + inizService.getConceptFromKey("report.vaccination.vaccinations").getConceptId()
+				        + " and value_coded =" + conceptService.getConceptByUuid(vacName).getConceptId()
+				        + ") AND concept_id ="
+				        + inizService.getConceptFromKey("report.vaccination.boostervaccinationSequenceNumberConcept")
+				                .getConceptId()
+				        + " AND value_numeric =" + lastIndex + ")";
+				
+				SqlCohortDefinition sql = new SqlCohortDefinition(sqlQuery);
+				sql.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+				sql.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+				vaccination.addRow(conceptService.getConceptByUuid(vacName).getDisplayString() + " " + lastIndex, sql,
+				    parameterMappings);
+				
+			}
+			
+		}
+		
+		// ECV
+		String[] ecvList = inizService.getValueFromKey("report.vaccination.ecvList").split(",");
+		String ecvQuery = "SELECT DISTINCT person_id FROM obs where 1=1";
+		
+		for (String member : ecvList) {
+			String[] bit = member.split(":");
+			String lastOne = bit[bit.length - 1];
+			if (!NumberUtils.isNumber(lastOne)) {
+				ecvQuery = ecvQuery
+				        + " AND person_id IN (SELECT person_id FROM obs where obs_group_id IN (SELECT obs_group_id FROM obs where concept_id="
+				        + inizService.getConceptFromKey("report.vaccination.vaccinations").getConceptId()
+				        + " and value_coded=" + conceptService.getConceptByUuid(member).getConceptId()
+				        + ") AND concept_id = "
+				        + inizService.getConceptFromKey("report.vaccination.vaccinationDate").getConceptId()
+				        + " and value_datetime <= :onOrBefore )";
+				
+			} else {
+				int lastIndex = Integer.parseInt(lastOne);
+				String vacName = member.substring(0, member.lastIndexOf(":"));
+				
+				ecvQuery = ecvQuery
+				        + " AND person_id IN ( SELECT person_id FROM obs where obs_group_id IN ( SELECT obs_group_id FROM obs where obs_group_id IN ( SELECT obs_group_id FROM obs where concept_id = "
+				        + inizService.getConceptFromKey("report.vaccination.vaccinations").getConceptId()
+				        + " and value_coded = " + conceptService.getConceptByUuid(vacName).getConceptId()
+				        + " ) AND concept_id = "
+				        + inizService.getConceptFromKey("report.vaccination.vaccinationDate").getConceptId()
+				        + " and value_datetime <= :onOrBefore ) AND concept_id ="
+				        + inizService.getConceptFromKey("report.vaccination.vaccinationSequenceNumberConcept").getConceptId()
+				        + " and value_numeric = " + lastIndex + "  AND :onOrAfter = :onOrAfter)";
```

Comment:
Not sure what this is doing? `AND :onOrAfter = :onOrAfter`

Reply:
An error is thrown when I don't do this replacement of the parameter...However I don't need this parameter for the ECV part....So I just did this hack